web/html/src/manager/maintenance/calendar/web-calendar.tsx

```diff
@@ -0,0 +1,262 @@
+          // Skip to next event that is not in current month
+          const firstEvent = moment.parseZone(events.data.filter(event =>
+            moment.parseZone(event.start).month() !== moment(currentDate).month())[0].start
+          );
+          getApi().gotoDate(firstEvent.format("YYYY-MM-DD"));
+          props.clearMessages();
+          setCurrentDate(getApi().currentDataManager.data.currentDate);
+        }
+      }).catch(props.responseError)
+  };
+
+  const onClickMonth = () => {
+    if (getApi().currentDataManager.data.currentViewType === "dayGridMonth") {
+      return;
+    }
+    if (needsUpdate(getApi().currentDataManager.data.currentDate)) {
+      getEvents("current");
+    }
+    getApi().changeView("dayGridMonth", getApi().currentDataManager.data.currentDate);
+  };
+
+  const onClickDay = () => {
+    getApi().changeView("timeGridDay", getApi().currentDataManager.data.currentDate);
+  };
+
+  // Switch to the day view of the clicked date
+  const onDateClick = (info) => {
+    setCurrentDate(getApi().currentDataManager.data.currentDate);
+    getApi().changeView('timeGridDay', info.date);
+    if (needsUpdate(info.date)) {
+      getEvents("current");
+    }
+  };
+
+  // Get the date used to fetch events based on the operation
+  const getDate = (operation) => {
+    switch (operation) {
+      case "next":
+        return Date.parse(getApi().currentDataManager.state.dateProfile.currentRange.end);
+      case "back":
+        // currentRange.start returns the first day of the displayed month, but we want to end up in
+        // the previous month so we subtract one day in this case
+        return Date.parse(getApi().currentDataManager.state.dateProfile.currentRange.start) - 24 * 60 * 60 * 1000;
+      case "current":
+        return Date.parse(getApi().currentDataManager.data.currentDate);
+      default:
+        return Date.parse(getApi().currentDataManager.data.dateProfileGenerator.nowDate);
+    }
+  };
+
+  const navigateTo = (operation) => {
+    switch (operation) {
+      case "next":
+        getApi().next();
+        break;
+      case "back":
+        getApi().prev();
+        break;
+      case "current":
+        getApi().gotoDate(getApi().currentDataManager.data.currentDate);
+        break;
+      default:
+        getApi().today();
+    }
+  }
+
+  const needsUpdate = (date) => {
+    return (moment(date).month() !== moment(currentDate).month() ||
+      moment(date).year() !== moment(currentDate).year());
+  };
+
+  const skipToNextEvent = (date) => {
+    // We are only interested in events from the same month that are later or equal to 'date'
+    const filteredEvents = events.filter(event => (
+      moment.parseZone(event.start).month() === moment(currentDate).month() &&
+      moment.parseZone(event.start).date() >= moment(date).date())
+    );
+    // Only perform action if there is an event to go to
+    filteredEvents.length > 0 && getApi().gotoDate(filteredEvents[0].start);
```

Comment:
Not wrong per se, I would however recommend an `if` to make it easier to spot that a conditional function call is taking place here when skimming through the code.

Reply:
Haha I knew that one would come up. Although I expected Dario to be the one to complain about it.
Well I like that format better, since it takes less space, but no strong aversion against using an if instead :P